logbus/solvermon/vertexmon.go

```diff
@@ -37,111 +37,147 @@ type vertexMonitor struct {
 	isCanceled     bool
 }
 
-var reErrExitCode = regexp.MustCompile(`^(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
-var reErrNotFound = regexp.MustCompile(`^failed to calculate checksum of ref ([^ ]*): (.*)$`)
-var reHint = regexp.MustCompile(`^(?P<msg>.+?):Hint: .+`)
+var reErrExitCode = regexp.MustCompile(`(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
 
-func (vm *vertexMonitor) Write(dt []byte, ts time.Time, stream int) (int, error) {
-	if stream == BuildkitStatsStream {
-		stats, err := vm.ssp.Parse(dt)
+func getExitCode(errString string) uint64 {
+	if reErrExitCode.MatchString(errString) {
+		matches, _ := stringutil.NamedGroupMatches(errString, reErrExitCode)
```

Comment:
this feels a bit odd, it looks like the `reErrExitCode` regex is being called twice (once in `reErrExitCode.MatchString(errString)`, then a second time inside the `NamedGroupMatches` function).

is it possible to only evaluate the regex once?

Reply:
Yeah, good point. Fixed.